File: TD4/app/src/main/java/com/example/newslist/DetailsActivity.java
package com.example.newslist;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class DetailsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_details);
        setTitle(getLocalClassName());

        Button ok_Btn = (Button)findViewById(R.id.ok_button);
        TextView login_Text = (TextView)findViewById(R.id.login_name);

        NewsListApplication app = (NewsListApplication) getApplicationContext();
        String login = app.getLogin();

        login_Text.setText(login);

        ok_Btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(DetailsActivity.this, NewsActivity.class);
                startActivity(intent);
            }
        });
    }

    @Override
    public void onBackPressed() {
        Intent intent = new Intent(DetailsActivity.this, NewsActivity.class);
        startActivity(intent);
    }
}

File: TD4 - Copie/app/src/main/java/com/example/newslist/DetailsActivity.kt
package com.example.newslist

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class DetailsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_details)
        title = localClassName
        val ok_Btn = findViewById<View>(R.id.ok_button) as Button
        val login_Text = findViewById<View>(R.id.login_name) as TextView
        val app = applicationContext as NewsListApplication
        val login = app.login
        login_Text.text = login
        ok_Btn.setOnClickListener {
            val intent = Intent(this@DetailsActivity, NewsActivity::class.java)
            startActivity(intent)
        }
    }

    override fun onBackPressed() {
        val intent = Intent(this@DetailsActivity, NewsActivity::class.java)
        startActivity(intent)
    }
}